### DISCUSSION TITLE
Generate OpenAPI schemas using Cue export

### DISCUSSION THREAD

**Discussion by aranw** [cue-lang/cue#1079] [Q&A]
I'm trying to work on some working examples of how I might use Cue for generating out various schema formats such as OpenAPI, Protobufs and maybe more in the future.

At the moment I'm trying to work on which can be found here https://github.com/aranw/cue-lang-action-test

I'm currently trying to run `cue export -I . --out openapi api/announcements/request.cue` to generate out an OpenAPI schema that inherits different definitions from other folders but soon as I've split it out to other folders it stops working. I've tried reading the documentation and the commands initially trying the `-I` flag but I've been unable to get this working. 

Is anyone able to help me work out what I need to run to get this working?

**Comment by mpvl** [comment]
This `-I` option is only for protobufs. You do not have to use it here.

What is probably going on is that some validation fails as the types are spread across packages. This is either a bug, or the error message should be more specific as to how to resolve it. Feel free to convert this into an issue.
